File: C++/Hard/WrongAnswer/number-of-different-subsequences-gcds_0.cpp
class Solution {
public:
    int countDifferentSubsequenceGCDs(vector<int>& nums) {
        
        int n = nums.size();
        int mx = *max_element(nums.begin(), nums.end());
        vector<int> cnt(mx + 1, 0);
        for (int i = 0; i < n; i++) {
            cnt[nums[i]]++;
        }
        int ans = 0;
        for (int i = 1; i <= mx; i++) {
            if (cnt[i] > 0) {
                ans++;
            }
            int g = 0;
            for (int j = i + i; j <= mx; j += i) {
                if (cnt[j] > 0) {
                    if (g == 0) {
                        g = j;
                    } else {
                        g = __gcd(g, j);
                    }
                }
            }
            if (g > 0 && __gcd(g, i) == i) {
                ans--;
            }
        }
        return ans;
    }
};

File: Python3/Hard/WrongAnswer/number-of-different-subsequences-gcds_0.py
class Solution:
    def countDifferentSubsequenceGCDs(self, nums: List[int]) -> int:
        
        def gcd(a, b):
            if a < b:
                a, b = b, a
            while b != 0:
                a, b = b, a % b
            return a
        nums = set(nums)
        res = 0
        for i in range(1, 2*10**5+1):
            if all(gcd(i, j) == i for j in nums):
                res += 1
        return res